src/File.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace React\Http;
+
+use React\Stream\ReadableStreamInterface;
+
+class File implements FileInterface
+{
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var string
+     */
+    protected $filename;
+
+    /**
+     * @var string
+     */
+    protected $type;
+
+    /**
+     * @var ReadableStreamInterface
+     */
+    protected $stream;
+
+    /**
+     * @param string $name
+     * @param string $filename
+     * @param string $type
+     * @param ReadableStreamInterface $stream
+     */
+    public function __construct($name, $filename, $type, ReadableStreamInterface $stream)
+    {
+        $this->name = $name;
+        $this->filename = $filename;
+        $this->type = $type;
+        $this->stream = $stream;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
```

Comment:
This is how you get it from request body.
Imagine following:
```html
<input type="file" name="uploaded_file">
```

What you'll get in request body is:
```
Content-Disposition: form-data; name="uploaded_file"; filename="picture.jpg"
Content-Type: image/jpeg
```

And how you get it normally in PHP is `$_FILES['uploaded_file']`.

So yes, `name` is essential here. And I agree, that PhpDoc might have some sort of comment to clarify this, even though for me (since I have own implementation of multipart parser) this is very clear already.